devel/make/BUILD

```diff
@@ -1,3 +1,32 @@
+# Before starting, see if any previous version of make exists and
+# if it works correctly.  Basically this should work even if
+# you've made a complete mess of your system and make no longer
+# functions.
+#
+# Also deal with the (quite unusual) circumstance of guile being
+# requested, but no longer working for whatever reason.  Someone
+# reported it breaking upon upgrade of libffi, for example.
+
+/usr/bin/make > /dev/null 2>& 1
+
+if [[ $? -eq 127 ]]  # this means make's gone missing or can't load
+then
+    case "$OPTS" in
+        *with-guile*)
+            /usr/bin/guile > /dev/null 2>&1
+            if [[ $? -eq 127 ]]  # and this means guile's broken (or missing)
+            then
+                # even though user asked for guile, don't use it this time round
+                OPTS=" --disable-dependency-tracking --without-guile "
+            fi
+        ;;
+
+        *)
+            OPTS+=" --disable-dependency-tracking "
```

Comment:
If this flag disables building with system make, is there any reason to avoid using it *everytime* to simplify and ensure the logic does not break without us noticing?

Reply:
It doesn't do that, it's just that in the absence of a working make already installed, configure doesn't complete without this flag.  Apart from that, I don't really know what it does, I'm just following the advice that the configure script gave me.

Code to deal with a broken make was already present in the BUILD, it was just ineffective because the BUILD script never got to it.